pysaliency/filter_datasets.py

```diff
@@ -232,3 +232,55 @@ def filter_stimuli_by_size(stimuli, fixations, size=None, sizes=None):
     indices = [i for i in range(len(stimuli)) if stimuli.sizes[i] in sizes]
 
     return create_subset(stimuli, fixations, indices)
+
+
+def filter_scanpaths_by_attribute(scanpaths:  FixationTrains, attribute_name, attribute_value, invert_match=False):
+    """Filter Scanpaths by values of scanpath attribute (fixation_trains.scanpath_attributes)"""
+
+    mask = scanpaths.scanpath_attributes[attribute_name] == attribute_value
+    if invert_match is True:
+        mask = ~mask
+    if mask.ndim>1:
+        mask = np.all(mask, axis=1)
```

Comment:
wouldn't this need to come before the `invert_match`? Right now, when `invert_match=False`, you're requiring _all_ entries to be different from the value.

Reply:
I am not sure what you mean by, requiring all entries to be difference from the value. In case of multi dim attribute I want value in each dimension to be equal, it can work before the invert_match if condition as well.

Reply:
no, it can't :). Let's assume one entry differs, then it shouldn't match (which it doesn't). But that also means that it should be a match with `invert_match=True`. But that's not the case:

1. original mask for that attribute value would be e.g. `[True, True, False, True]`
2. since `invert_match=True`, the mask is inverted to `[False, False, True, False]`
3. now `np.all` returns `False`, but it would need to return `True`.

If you do the accumulation before the `invert_match`, `np.all` would have worked (`[True, True, False, True] -> False -> True). After the `invert_match`, you would need to use `np.any` for `invert_match=False`. This is basically an application of `not (A and B) == (not A) or (not B)`

Reply:
ahhh okay now I get it. I will move it before the `invert_match` if condition.